x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/droppable.ts

```diff
@@ -159,6 +177,59 @@ export function onDrop(props: DatasourceDimensionDropHandlerProps<IndexPatternPr
   const hasData = Object.values(state.layers).some(({ columns }) => columns.length);
   trackUiEvent(hasData ? 'drop_non_empty' : 'drop_empty');
   setState(mergeLayer({ state, layerId, newLayer }));
-
   return true;
+};
+
+export function onDrop(props: DatasourceDimensionDropHandlerProps<IndexPatternPrivateState>) {
+  const operationSupportMatrix = getOperationSupportMatrix(props);
+  const { setState, state, droppedItem, columnId, layerId, groupId, isNew } = props;
+
+  if (!isDraggedOperation(droppedItem)) {
+    return onFieldDrop({
+      columnId,
+      setState,
+      state,
+      layerId,
+      droppedItem,
+      operationSupportMatrix,
+    });
+  }
+  const isExistingFromSameGroup =
+    droppedItem.groupId === groupId && droppedItem.columnId !== columnId && !isNew;
+
+  // reorder in the same group
+  if (isExistingFromSameGroup) {
```

Comment:
In the old version reorder was detected using `props.isReorder` which seems simpler. Is there a reason we can't do that anymore?

Reply:
I just wanted to get rid of a prop that was very specific to reordering use case. In the future, I am planning on passing `dropType` to this function (which will be evaluated in `getDropTypes` which will replace `canHandleDrop` and then passed to `onDrop` function) – I hope it's ok with you to leave it as it is right now, but if you feel strongly about the other implementation let me know and I'll revert it!

Reply:
Sure, sounds good to me.